File: mars-ui/src/api/douyin-product.ts
import { request } from '@/utils/request'

// 抖音商品类型定义（本地数据库）
export interface DouyinProduct {
  id: number
  ownerId: string
  productId: string | null
  outerProductId: string | null
  name: string
  pic: string | null
  description: string | null
  price: number | null
  originalPrice: number | null
  costPrice: number | null
  stockNum: number | null
  status: number
  statusName: string
  auditStatus: number
  auditStatusName: string
  auditRemark: string | null
  categoryLeafId: number | null
  productType: number | null
  productTypeName: string
  deliveryMethod: number | null
  presellType: number | null
  presellConfigLevel: number | null
  presellDeliveryType: number | null
  supply7dayReturn: number | null
  createTime: string
  updateTime: string | null
  publishTime: string | null
  // 物流信息
  weight: number | null
  weightUnit: number | null
  deliveryDelayDay: number | null
  freightId: number | null
  // 限购设置
  maximumPerOrder: number | null
  minimumPerOrder: number | null
  limitPerBuyer: number | null
  // 库存扣减方式
  reduceType: number | null
  // 支付方式
  payType: number | null
  // 预售延迟
  presellDelay: number | null
  // 推荐备注
  recommendRemark: string | null
  // 规格
  specName: string | null
  spuId: number | null
  // 其他
  commit: boolean | null
  mobile: string | null
  remark: string | null
}

// 抖音平台商品类型定义（从抖音API查询）
export interface DouyinProductV2 {
  productId: number
  name: string
  img: string | null
  description: string | null
  recommendRemark: string | null
  status: number
  checkStatus: number
  productType: number
  payType: number | null
  marketPrice: number | null // 单位：分
  discountPrice: number | null // 单位：分
  outerProductId: string | null
  freightId: number | null
  mobile: string | null
  createTime: number | null // 时间戳（秒）
  updateTime: number | null // 时间戳（秒）
  sellNum: number | null
  canCombine: boolean | null
  isSecondHandDigital: boolean | null
  isPackageProduct: boolean | null
  specPrices: SpecPriceItem[] | null
  categoryDetail: CategoryDetail | null
  shopCategory: ShopCategory | null
}

export interface SpecPriceItem {
  id: number
  code: string | null
  barcodes: string[] | null
}

export interface CategoryDetail {
  firstCid: number | null
  secondCid: number | null
  thirdCid: number | null
  fourthCid: number | null
  firstCname: string | null
  secondCname: string | null
  thirdCname: string | null
  fourthCname: string | null
}

export interface ShopCategory {
  leafCategoryIds: number[] | null
}

// 商品保存请求
export interface DouyinProductSaveReq {
  id?: number
  ownerId?: string
  productId?: string
  outerProductId?: string
  name?: string
  pic?: string
  description?: string
  categoryLeafId?: number | null
  price?: number | null
  originalPrice?: number | null
  costPrice?: number | null
  stockNum?: number | null
  deliveryMethod?: number | null
  weight?: number | null
  weightUnit?: number | null
  deliveryDelayDay?: number | null
  presellType?: number | null
  presellConfigLevel?: number | null
  presellDeliveryType?: number | null
  supply7dayReturn?: number | null
  remark?: string
  status?: number
  // 限购设置
  maximumPerOrder?: number | null
  minimumPerOrder?: number | null
  limitPerBuyer?: number | null
  // 规格
  specName?: string
  spuId?: number | null
  // 其他
  commit?: boolean
  mobile?: string
  // 支付
  payType?: number | null
  // 库存
  reduceType?: number | null
  // 运费
  freightId?: number | null
  // 预售
  presellDelay?: number | null
  // 推荐备注
  recommendRemark?: string
  // 商品类型
  productType?: number | null
}

// 商品查询参数（本地数据库）
export interface DouyinProductQueryReq {
  ownerId?: string
  productId?: string
  outerProductId?: string
  name?: string
  categoryLeafId?: number | null
  status?: number | null
  auditStatus?: number | null
  page?: number
  pageSize?: number
}

// 商品查询参数（抖音平台）
export interface DouyinProductListV2Req {
  ownerId?: string
  name?: string
  productId?: string[]
  skuCodes?: string[]
  status?: number | null
  checkStatus?: number | null
  productType?: number | null
  storeId?: number | null
  startTime?: number | null // 时间戳（秒）
  endTime?: number | null // 时间戳（秒）
  updateStartTime?: number | null
  updateEndTime?: number | null
  page?: number
  size?: number
  useCursor?: boolean
  cursorId?: string
  canCombineProduct?: boolean
  needRectificationInfo?: boolean
  needCheckOut?: boolean
}

// 分页响应
export interface PageResult<T> {
  list: T[]
  total: number
  page: number
  pageSize: number
}

// 抖音平台商品列表响应
export interface DouyinProductListV2Rsp {
  data: DouyinProductV2[]
  total: number
  page: number
  size: number
  cursorId: string | null
}

// 抖音商品 API
export const douyinProductApi = {
  // 分页查询（本地数据库）
  page(params: { page: number; pageSize: number } & DouyinProductQueryReq) {
    return request<PageResult<DouyinProduct>>({ url: '/biz/douyin-product/page', method: 'get', params })
  },

  // 列表查询（本地数据库）
  list(params: DouyinProductQueryReq) {
    return request<{ list: DouyinProduct[]; total: number }>({ url: '/biz/douyin-product/list', method: 'get', params })
  },

  // 从抖音平台查询商品列表
  listProductV2(params: DouyinProductListV2Req) {
    return request<DouyinProductListV2Rsp>({ url: '/biz/douyin-product/product/list', method: 'get', params })
  },

  // 获取详情
  getInfo(id: number) {
    return request<DouyinProduct>({ url: `/biz/douyin-product/${id}`, method: 'get' })
  },

  // 新增
  add(data: DouyinProductSaveReq) {
    return request<DouyinProduct>({ url: '/biz/douyin-product', method: 'post', data })
  },

  // 更新
  update(id: number, data: DouyinProductSaveReq) {
    return request<DouyinProduct>({ url: `/biz/douyin-product/${id}`, method: 'put', data })
  },

  // 保存（新增或更新）
  save(data: DouyinProductSaveReq) {
    return request<DouyinProduct>({ url: '/biz/douyin-product/save', method: 'post', data })
  },

  // 删除
  delete(ids: number[]) {
    return request({ url: `/biz/douyin-product/${ids.join(',')}`, method: 'delete' })
  },

  // 上架
  onSale(id: number) {
    return request({ url: `/biz/douyin-product/${id}/on-sale`, method: 'put' })
  },

  // 下架
  offSale(id: number) {
    return request({ url: `/biz/douyin-product/${id}/off-sale`, method: 'put' })
  },

  // 提交审核
  submitAudit(id: number) {
    return request({ url: `/biz/douyin-product/${id}/submit-audit`, method: 'put' })
  },

  // 根据外部 ID 查询
  getByOwnerIdAndProductId(ownerId: string, productId: string) {
    return request<DouyinProduct>({
      url: '/biz/douyin-product/external',
      method: 'get',
      params: { ownerId, productId }
    })
  },
}

File: mars-core/mars-biz/src/main/java/com/mars/biz/entity/DouyinProduct.java
package com.mars.biz.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 抖音商品表
 *
 * @author Mars
 * @date 2026-04-18
 */
@Data
@TableName("douyin_product")
public class DouyinProduct implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 主键 ID
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 店铺所有者 ID（关联店铺表）
     */
    private String ownerId;

    /**
     * 抖音商品 ID
     */
    private String productId;

    /**
     * 外部商品 ID（商家自定义）
     */
    private String outerProductId;

    /**
     * 外部商品 ID（数字类型）
     */
    private Long outProductId;

    // ==================== 基本信息 ====================

    /**
     * 商品类型：1-普通商品，2-电子面单商品
     */
    private Long productType;

    /**
     * 叶子类目 ID
     */
    private Long categoryLeafId;

    /**
     * 商品名称
     */
    private String name;

    /**
     * 商品主图 URL
     */
    private String pic;

    /**
     * 商品描述
     */
    private String description;

    /**
     * 推荐备注
     */
    private String recommendRemark;

    // ==================== 价格库存 ====================

    /**
     * 支付方式：0-全款，1-定金 + 尾款
     */
    private Long payType;

    /**
     * 商品价格（元）
     */
    private BigDecimal price;

    /**
     * 原价/划线价（元）
     */
    private BigDecimal originalPrice;

    /**
     * 成本价（元）
     */
    private BigDecimal costPrice;

    /**
     * 库存数量
     */
    private Integer stockNum;

    /**
     * 库存扣减方式：1-付款减库存，2-发货减库存
     */
    private Long reduceType;

    // ==================== 物流信息 ====================

    /**
     * 配送方式：1-快递，2-上门取件，3-商家自配
     */
    private Integer deliveryMethod;

    /**
     * 运费模板 ID
     */
    private Long freightId;

    /**
     * 商品重量
     */
    private BigDecimal weight;

    /**
     * 重量单位：1-kg，2-g，3-jin
     */
    private Long weightUnit;

    /**
     * 发货延迟天数
     */
    private Long deliveryDelayDay;

    // ==================== 预售信息 ====================

    /**
     * 预售类型：0-非预售，1-全款预售，2-定金预售
     */
    private Long presellType;

    /**
     * 预售延迟天数
     */
    private Long presellDelay;

    /**
     * 预售结束时间（时间戳）
     */
    private String presellEndTime;

    /**
     * 预售配置等级
     */
    private Long presellConfigLevel;

    /**
     * 预售发货类型
     */
    private Long presellDeliveryType;

    // ==================== 规格 SKU ====================

    /**
     * 规格名称
     */
    private String specName;

    /**
     * 规格信息（JSON）
     */
    private String specs;

    /**
     * 规格价格（JSON）
     */
    private String specPrices;

    /**
     * 规格图片（JSON）
     */
    private String specPic;

    // ==================== 限购设置 ====================

    /**
     * 每单最大购买数量
     */
    private Long maximumPerOrder;

    /**
     * 每单最小购买数量
     */
    private Long minimumPerOrder;

    /**
     * 每人限购数量
     */
    private Long limitPerBuyer;

    // ==================== 商品属性 ====================

    /**
     * 商品属性（JSON）
     */
    private String productFormatNew;

    /**
     * SPU ID
     */
    private Long spuId;

    /**
     * 标准品牌 ID
     */
    private Long standardBrandId;

    /**
     * 资质列表（JSON）
     */
    private String qualityList;

    // ==================== 其他设置 ====================

    /**
     * 跨境购类目
     */
    private String cdfCategory;

    /**
     * 关联商品 ID 列表
     */
    private String assocIds;

    /**
     * 联系电话
     */
    private String mobile;

    /**
     * 是否支持 7 天无理由：0-不支持，1-支持
     */
    private Long supply7dayReturn;

    /**
     * 是否提交审核
     */
    private Boolean commit;

    /**
     * 商家备注
     */
    private String remark;

    /**
     * 是否需要 checkout
     */
    private Boolean needCheckOut;

    /**
     * 是否需要充值模式
     */
    private Boolean needRechargeMode;

    // ==================== 发货相关 ====================

    /**
     * 预约发货天数
     */
    private Long appointDeliveryDay;

    /**
     * 第三方链接
     */
    private String thirdUrl;

    /**
     * 扩展信息（JSON）
     */
    private String extra;

    /**
     * 商品来源
     */
    private String src;

    // ==================== O2O 相关 ====================

    /**
     * 门店资源（JSON）
     */
    private String poiResource;

    /**
     * 汽车 VIN 码
     */
    private String carVinCode;

    // ==================== 图片相关 ====================

    /**
     * 白底图 URL
     */
    private String whiteBackgroundPicUrl;

    /**
     * 长图 URL
     */
    private String longPicUrl;

    // ==================== 账号相关 ====================

    /**
     * 账号模板 ID
     */
    private String accountTemplateId;

    // ==================== 状态字段 ====================

    /**
     * 状态：0-草稿，1-审核中，2-已上架，3-已下架，4-审核驳回，5-已删除
     */
    private Integer status;

    /**
     * 审核状态：0-待审核，1-审核通过，2-审核驳回
     */
    private Integer auditStatus;

    /**
     * 审核驳回原因
     */
    private String auditRemark;

    // ==================== 时间字段 ====================

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    /**
     * 上架时间
     */
    private LocalDateTime publishTime;

    /**
     * 审核时间
     */
    private LocalDateTime auditTime;

    // ==================== 状态常量 ====================

    /**
     * 状态：草稿
     */
    public static final int STATUS_DRAFT = 0;

    /**
     * 状态：审核中
     */
    public static final int STATUS_AUDITING = 1;

    /**
     * 状态：已上架
     */
    public static final int STATUS_ON_SALE = 2;

    /**
     * 状态：已下架
     */
    public static final int STATUS_OFF_SALE = 3;

    /**
     * 状态：审核驳回
     */
    public static final int STATUS_REJECTED = 4;

    /**
     * 状态：已删除
     */
    public static final int STATUS_DELETED = 5;

    /**
     * 审核状态：待审核
     */
    public static final int AUDIT_STATUS_PENDING = 0;

    /**
     * 审核状态：审核通过
     */
    public static final int AUDIT_STATUS_APPROVED = 1;

    /**
     * 审核状态：审核驳回
     */
    public static final int AUDIT_STATUS_REJECTED = 2;

    /**
     * 获取状态名称
     */
    @TableField(exist = false)
    private String statusName;

    /**
     * 获取审核状态名称
     */
    @TableField(exist = false)
    private String auditStatusName;

    public String getStatusName() {
        if (status == null) {
            return "未知";
        }
        return switch (status) {
            case STATUS_DRAFT -> "草稿";
            case STATUS_AUDITING -> "审核中";
            case STATUS_ON_SALE -> "已上架";
            case STATUS_OFF_SALE -> "已下架";
            case STATUS_REJECTED -> "审核驳回";
            case STATUS_DELETED -> "已删除";
            default -> "未知";
        };
    }

    public String getAuditStatusName() {
        if (auditStatus == null) {
            return "未知";
        }
        return switch (auditStatus) {
            case AUDIT_STATUS_PENDING -> "待审核";
            case AUDIT_STATUS_APPROVED -> "审核通过";
            case AUDIT_STATUS_REJECTED -> "审核驳回";
            default -> "未知";
        };
    }
}
